README.md

```diff
@@ -1 +1,177 @@
+
+To test downloading a certificate:
+
+1. Backup your aws credentials file and change the default credentials to the serverless-s3-local ones (`S3RVER`, `S3RVER`)
+1. create a folder in the root of the project called `buckets`
+1. Run `aws --endpoint http://localhost:4569 s3api put-object --bucket cvs-cert-develop --key <your certificate name> --body <location of your certificate including filename>` to put a copy of your certificate into the S3 bucket which is in the `buckets` folder you created.
+1. Change the metadata of your item in the bucket so that the mime type is `application/octet-stream`
+1. Add
+
+```
+{
+  s3ForcePathStyle: true,
+  endpoint: new AWS.Endpoint('http://localhost:4569'),
+}
+```
+
+to the initialisation of the S3 client in the API index
+
+1. Use Postman to request the certificate using the VIN and certificate number in the name of your certificate. E.g. http://localhost:3001/local/document-retrieval?vinNumber=T12765432&testNumber=W01A00229 for a file with a key of `W01A00229_T12765432.pdf`
+1. Convert the base64 encoded string to a PDF using a service, e.g. https://base64.guru/converter/decode/pdf
+
+You should be able to see the certificate on your screen.
+
+### Lambda locally
+
+Serverless can invoke lambda functions locally which provide a close experience to the real service if you decide not use the offline mode. `events` and `paths` can be found under `/local` folder.
+For further details using lambda locally please refer to the [serverless documentation](https://www.serverless.com/framework/docs/providers/aws/cli-reference/invoke-local/).
+
+### Debugging
+
+Existing configuration to debug the running service has been made available for vscode, please refer to `.vscode/launch.json` file. Serverless offline will be available on port `:4000`. 2 jest configurations are also provided which will allow to run a test or multiple tests.
+
+For further information about debugging, please refer to the following documentation:
+
+- [Run-a-function-locally-on-source-changes](https://github.com/serverless-heaven/serverless-webpack#run-a-function-locally-on-source-changes)
+
+- [VSCode debugging](https://github.com/serverless-heaven/serverless-webpack#vscode-debugging)
+
+- [Debug process section](https://www.serverless.com/plugins/serverless-offline#usage-with-webpack)
+
+## Testing
+
+[json-serverless](https://github.com/pharindoko/json-serverless) has been added to the repository should we wish to mock external services during development and can be used in conjunction with the `test` environment.
+
+### Unit
+
+Jest is used for unit testing.
+Please refer to the [Jest documentation](https://jestjs.io/docs/en/getting-started) for further details.
+
+### Integration
+
+To be added and customised depending on needs, supertest is used but we could be looking at other packages such as nock, ts-mockito, typemoq, wiremock, etc.. or testing (pactjs, hoverfly, mockserver, etc..)
+
+## Infrastructure
+
```

Comment:
I don't know if we have a low level design for this service (architecture diagram, class diagram, etc..) if yes we could add it otherwise remove if we don't have any?